app/controllers/OtherReasonController.scala

```diff
@@ -200,9 +199,49 @@ class OtherReasonController @Inject()(whenDidBecomeUnablePage: WhenDidBecomeUnab
       )
   }
 
-  def onPageLoadForUploadComplete(mode: Mode): Action[AnyContent] = (authorise andThen dataRequired) {
+  def onPageLoadForUploadComplete(mode: Mode): Action[AnyContent] = (authorise andThen dataRequired).async {
     implicit request => {
-      Ok("successful upload page")
+      val formProvider: Form[String] = FormProviderHelper.getSessionKeyAndAttemptToFillAnswerAsString(
+        YouHaveUploadedFilesForm.youHaveUploadedForm,
+        SessionKeys.nextFileUpload
+      )
+      val radioOptionsToRender: Seq[RadioItem] = RadioOptionHelper.yesNoRadioOptions(formProvider)
+      val postAction = controllers.routes.OtherReasonController.onSubmitForUploadComplete()
+      val journeyId = request.session.get(SessionKeys.journeyId).get
+      upscanService.getAmountOfFilesUploadedForJourney(journeyId).map(
+        filesUploaded => {
+          Ok(youHaveUploadedFilesPage(formProvider, radioOptionsToRender, postAction, filesUploaded))
+        }
+      )
+    }
+  }
+
+  def onSubmitForUploadComplete(mode: Mode): Action[AnyContent] = (authorise andThen dataRequired).async {
+    implicit request => {
+      YouHaveUploadedFilesForm.youHaveUploadedForm.bindFromRequest.fold(
```

Comment:
The form should not appear if the user has uploaded 5 documents that are now in the READY status, so we need to handle that

Reply:
Handled in the OtherReasonController.onPageLoadForUploadComplete  page YouHaveUploadedFilesPage.scala.html